### PR TITLE
Join for visits

### PR DESCRIPTION
**Pull request by benwmx**
## Details 
In this pull request I added the following :
* Create vets table.
* Create specializations table.
* Create visits table.
* Insert vets Data.
* Insert specializations data.
* Insert visits data.
* Queries for analytical questions.

## ScreenShots
Lpr

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -24,4 +24,24 @@ CREATE TABLE species (
...
+    vet_id INT REFERENCES vets(id),
+    species_id INT REFERENCES species(id),
+    PRIMARY KEY (species_id, vet_id)
+);
+
+CREATE TABLE visits (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    animal_id INT REFERENCES animals(id),
+    vet_id INT REFERENCES vets(id),
+    date_of_visit DATE
```
Comment (Emmyn5600) [inline schema.sql:46]: Optional suggestions

- [ ] We don't need the id column for this table, only `animal_id and vet_id` are enough.

File: queries.sql [hunk 2]
```diff
@@ -72,3 +72,44 @@ SELECT owners.full_name, COUNT(animals.name) number_of_animals
...
+FROM vets JOIN (visits JOIN animals ON visits.animal_id = animals.id) ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez'
+GROUP BY vets.name;
+
+SELECT vets.name vets_name, species.name species_name
+FROM vets LEFT JOIN (specializations JOIN species ON specializations.species_id = species.id) ON vets.id = specializations.vet_id;
+
+SELECT vets.name vet_name, animals.name animal_name, date_of_visit
+FROM vets JOIN (visits JOIN animals ON visits.animal_id = animals.id) ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez' AND date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (Emmyn5600) [inline queries.sql:93]: Optional suggestion

- [ ] Your solution works well, however, I would recommend using LEFT JOIN here, refer to the below example.

``` SELECT animals.name, vistis.visit_date
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez'
  AND visit_date > '2020-04-01' AND visit_date < '2020-08-30'```

**Review by Emmyn5600 — APPROVED** [review]
Hi @benwmx 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada:

 #### Optional suggestions
 
 - Check the comment under the review.
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Comment by benwmx** [comment]
Thanks @Emmyn5600  For your time reviewing my code, and also for your suggestions.

**Pull request merged**